client.go

```diff
@@ -15,11 +15,15 @@ type Client struct {
 // Do sends a request and waits for a response.
 // If the response frame doesn't arrive on time, an error is returned.
 func (c *Client) Do(req *Request) (*Response, error) {
+	respChan := make(chan can.WaitResponse)
+	go func() {
+		response := <-can.Wait(c.Bus, req.ResponseID, c.Timeout)
+		respChan <- response
+	}()
+	time.Sleep(time.Millisecond * 15)
 	if err := c.Bus.Publish(req.Frame.CANFrame()); err != nil {
 		return nil, err
 	}
-
-	resp := <-can.Wait(c.Bus, req.ResponseID, c.Timeout)
-
+	resp := <-respChan
```

Comment:
I'm rather new in go, but is here still a timeout? It waits infinite at this point, I guess because it missed the response and the channel waits for data

Reply:
Yes, `can.Wait(...)` expects a timeout as the third argument.

Reply:
> Thanks for your contribution.
> 
> Your approach to fix this issue seems reasonable.
> But the introduction of a sleep call adds a delay to the method.
> 
> Why not publish the can frame in the new go routine like this?
> 

Just tried this, seems to work. Also more elegant solution than mine.

Reply:
> > Thanks for your contribution.
> > Your approach to fix this issue seems reasonable.
> > But the introduction of a sleep call adds a delay to the method.
> > Why not publish the can frame in the new go routine like this?
> 
> Just tried this, seems to work. Also more elegant solution than mine.
Unfortunately it still fails, just less often.